### PR TITLE
Task #5258 - Fix i18n (user comment book)

### PR DESCRIPTION
**Pull request by taylq**
https://edu-redmine.framgia.vn/issues/5258?issue_count=20&issue_position=3&next_issue_id=5249&prev_issue_id=5260

### CODE REVIEW

**Comment by taylq** [comment]
ready

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
+class CommentsController < ApplicationController
+  before_action :correct_user, only: :destroy
+
+  def create
+    @comment = current_user.comments.build comment_params
```
Comment (huuphuc93) [inline app/controllers/comments_controller.rb]:
```suggestion
    @comment = current_user.comments.build comment_params.merge(book_id: params[:book_id])
```

File: app/models/comment.rb [hunk 2]
```diff
@@ -1,4 +1,8 @@
 class Comment < ApplicationRecord
   belongs_to :book
   belongs_to :user
+
+  validates :content, presence: true
+
+  default_scope -> {order(created_at: :desc)}
```
Comment (huuphuc93) [inline app/models/comment.rb]: k dùng default_scope

File: app/views/books/show.html.erb [hunk 3]
```diff
@@ -44,6 +45,14 @@
         <div class="col-sm-8" style="text-align: center">
           <h2><%= @book.name %></h2>
           <h6><%= @book.content %></h6>
+          <div id="load_comment-<%= @book.id %>">
+            <%= render @comments %>
+          </div>
+          <% if logged_in? %>
+            <div id="comment_form">
+              <%= render "comments/form", book: @book, user: current_user %>
```
Comment (huuphuc93) [inline app/views/books/show.html.erb]: k cần truyền current_user

**Comment by huuphuc93** [comment]
commented

**Comment by taylq** [comment]
ready

**Comment by huuphuc93** [comment]
commented

**Comment by taylq** [comment]
ready

**Pull request merged**